addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4048,7 +4048,6 @@ var SnippetsMenu = Widget.extend({
 
         this._$removeFormatButton = this._$removeFormatButton || this.options.wysiwyg.toolbar.$el.find('#removeFormat');
         $title.append(this._$removeFormatButton);
-        this.options.wysiwyg.toolbar.$el.find('#table').remove();
```

Comment:
This is called by `SnippetsMenu.start`. So this might be the code that removes the table in case snippets is true...

Reply:
Mmhh... You're probably right, yes.